turbopack/crates/turbo-tasks-backend/src/backend/operation/mod.rs

```diff
@@ -631,6 +639,7 @@ impl<B: BackingStorage> TaskGuard for TaskGuardImpl<'_, B> {
         &self,
         ty: CachedDataItemType,
     ) -> impl Iterator<Item = (CachedDataItemKey, CachedDataItemValueRef<'_>)> {
+        self.check_access(ty.category());
```

Comment:
Because this debug assertion was missing, it was possible to call `iter()` on data that was not restored. This e. g. led to an empty dependents list, which breaks all kind of stuff.